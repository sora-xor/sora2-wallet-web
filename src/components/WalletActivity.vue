<template>
  <wallet-history :history="activity" />
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import { Getter } from 'vuex-class'

import WalletHistory from './WalletHistory.vue'

@Component({
  components: { WalletHistory }
})
export default class WalletActivity extends Vue {
  @Getter activity!: Array<any>

  mounted (): void {
  }
}
</script>
