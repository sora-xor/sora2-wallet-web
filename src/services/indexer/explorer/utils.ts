import { formatStringNumber } from '../../../util';

import type { SubqueryAssetEntity } from '../subquery/types';
import type { SubsquidAssetEntity } from '../subsquid/types';
import type { FiatPriceObject, UpdatesStream, PoolApyObject } from '../types';

export function parseAssetFiatPrice(entity: SubsquidAssetEntity | SubqueryAssetEntity): FiatPriceObject {
  const acc = {};
  const id = entity.id;
  const priceFPNumber = formatStringNumber(entity.priceUSD);
  const isPriceFinity = priceFPNumber.isFinity();
  if (isPriceFinity) {
    acc[id] = priceFPNumber.toCodecString();
  }
  return acc;
}

export function parsePriceStreamUpdate(entity: UpdatesStream): Nullable<FiatPriceObject> {
  if (!entity?.data) return null;

  const data = JSON.parse(entity.data);

  return Object.entries(data).reduce((acc, [id, price]) => {
    const priceFPNumber = formatStringNumber(price as string);
    const isPriceFinity = priceFPNumber.isFinity();
    if (isPriceFinity) {
      acc[id] = priceFPNumber.toCodecString();
    }
    return acc;
  }, {});
}

export function parseApyStreamUpdate(entity: UpdatesStream): PoolApyObject {
  const data = entity?.data ? JSON.parse(entity.data) : {};

  return Object.entries(data).reduce((acc, [id, apy]) => {
    const strategicBonusApyFPNumber = formatStringNumber(apy as string);
    const isStrategicBonusApyFinity = strategicBonusApyFPNumber.isFinity();
    if (isStrategicBonusApyFinity) {
      acc[id] = strategicBonusApyFPNumber.toCodecString();
    }
    return acc;
  }, {});
}
