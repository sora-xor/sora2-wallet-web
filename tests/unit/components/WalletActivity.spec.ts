import { shallowMount } from '@vue/test-utils';

import WalletActivity from '@/components/WalletActivity.vue';
import { useDescribe, localVue, i18n } from '../../utils';

useDescribe('WalletActivity.vue', WalletActivity, () => {
  it('should be rendered correctly', () => {
    const wrapper = shallowMount(WalletActivity, { localVue, i18n });

    expect(wrapper.element).toMatchSnapshot();
  });
});
