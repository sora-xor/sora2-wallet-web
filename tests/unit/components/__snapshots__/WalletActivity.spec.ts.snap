// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletActivity.vue should be rendered correctly 1`] = `<wallet-history-stub />`;
